src/coreclr/gc/vxsort/machine_traits.avx2.h

```diff
@@ -24,8 +24,28 @@
 #define d2s _mm256_castpd_ps
 
 namespace vxsort {
-extern const int8_t perm_table_64[128];
-extern const int8_t perm_table_32[2048];
+
+// We might read the last 8 bytes into a 128-bit vector for the purpose of permutation
+// Most compilers actually fuse the pair of instructions: _mm256_cvtepi8_epiNN + _mm_loadu_si128
+// into a single instruction that will not read more that 4/8 bytes..
+// vpmovsxb[dq] ymm0, dword [...], eliminating the 128-bit load completely and effectively
+// reading exactly 4/8 (depending if the instruction is vpmovsxbd or vpmovsxbq)
+// without generating an out of bounds read at all.
+// But, life is harsh, and we can't trust the compiler to do the right thing if it is not
```

Comment:
If I am reading this correctly, we are trusting the compiler to do the "right thing" when it does not have `address_sanitizer` feature. Why is that a good idea?

Reply:
The comment was copied from upstream [here](https://github.com/damageboy/vxsort-cpp/blob/096e967cd6a3be2b18d4baf1af81893d97fad7dd/vxsort/machine_traits.avx2.h#L23)

and the pattern of including the padding bytes conditionally was copied [here](https://github.com/dotnet/runtime/blob/4cf1383c8458945b7eb27ae5f57338c10ed25d54/src/coreclr/gc/vxsort/smallsort/avx2_load_mask_tables.cpp#L16)

By the time I was copying them, I didn't notice they contradict each other.

Perhaps we should remove the `#if !__has_feature(address_sanitizer)` to stay on the safe side.